app/models/ont/library_factory.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+
+# Ont namespace
+module Ont
+  # LibraryFactory
+  # The factory will create libraries in tubes from a given plate
+  class LibraryFactory
+    include ActiveModel::Model
+
+    validate :check_validation_errors,
+             :check_libraries,
+             :check_library_requests,
+             :check_container_materials
```

Comment:
I think you may be missing a validation for `@tubes` here.

Reply:
I am not :). If I add a check on tubes they get validated twice, hence I do not explicitly validate

Reply:
Maybe pop a comment in to help future developers (aka you) when they think the same thing I did